packages/aws-cdk-lib/aws-ec2/lib/nat.ts

```diff
@@ -399,6 +419,115 @@ class PrefSet<A> {
   }
 }
 
+/**
+ * Modern NAT provider which uses NAT Instances.
+ * The instance uses Amazon Linux 2023 as the operating system.
+ */
+export class NatInstanceProviderV2 extends NatProvider implements IConnectable {
+  private gateways: PrefSet<Instance> = new PrefSet<Instance>();
+  private _securityGroup?: ISecurityGroup;
+  private _connections?: Connections;
+
+  constructor(private readonly props: NatInstanceProps) {
+    super();
+
+    if (props.defaultAllowedTraffic !== undefined && props.allowAllTraffic !== undefined) {
+      throw new Error('Can not specify both of \'defaultAllowedTraffic\' and \'defaultAllowedTraffic\'; prefer \'defaultAllowedTraffic\'');
+    }
+
+    if (props.keyName && props.keyPair) {
+      throw new Error('Cannot specify both of \'keyName\' and \'keyPair\'; prefer \'keyPair\'');
+    }
+  }
+
+  public configureNat(options: ConfigureNatOptions) {
+    const defaultDirection = this.props.defaultAllowedTraffic ??
+      (this.props.allowAllTraffic ?? true ? NatTrafficDirection.INBOUND_AND_OUTBOUND : NatTrafficDirection.OUTBOUND_ONLY);
+
+    // Create the NAT instances. They can share a security group and a Role.
+    const machineImage = this.props.machineImage || new AmazonLinuxImage({ generation: AmazonLinuxGeneration.AMAZON_LINUX_2023 });
```

Comment:
Can we document why we are doing an `InstanceV2` here? I don't quite understand. This looks like the important change between the two. Does that mean that everything under `NatInstanceImage()` is old-generation and deprecated?

Either way, lets document clearly why we went down this path